steampy/login.py

```diff
@@ -117,11 +115,53 @@ def _assert_valid_credentials(login_response: Response) -> None:
             raise InvalidCredentials(login_response.json()['message'])
 
     def _perform_redirects(self, response_dict: dict) -> None:
-        parameters = response_dict.get('transfer_parameters')
+        parameters = response_dict.get('transfer_info')
         if parameters is None:
             raise Exception('Cannot perform redirects after login, no parameters fetched')
-        for url in response_dict['transfer_urls']:
-            self.session.post(url, parameters)
+        for pass_data in parameters:
+            pass_data['params']['steamID'] = response_dict['steamID']
+            self.session.post(pass_data['url'], pass_data['params'])
 
     def _fetch_home_page(self, session: Session) -> Response:
         return session.post(SteamUrl.COMMUNITY_URL + '/my/home/')
+
+    def _update_steam_guard(self, login_response: Response) -> bool:
+        client_id = login_response.json()["response"]["client_id"]
+        steamid = login_response.json()["response"]["steamid"]
+        request_id = login_response.json()["response"]["request_id"]
+        code_type = 3
+        code = guard.generate_one_time_code(self.shared_secret)
+
+        update_data = {
+            'client_id': client_id,
+            'steamid': steamid,
+            'code_type': code_type,
+            'code': code
+        }
+        response = self._api_call('POST', 'IAuthenticationService', 'UpdateAuthSessionWithSteamGuardCode',
+                                  params = update_data)
+        if response.status_code == 200:
+            self._pool_sessions_steam(client_id, request_id)
+            return True
+        else:
+            raise Exception('Cannot update steam guard')
+
+    def _pool_sessions_steam(self, client_id, request_id):
+        pool_data = {
+            'client_id': client_id,
+            'request_id': request_id
+        }
+        response = self._api_call('POST', 'IAuthenticationService', 'PollAuthSessionStatus', params = pool_data)
+        self.refresh_token = response.json()["response"]["refresh_token"]
+
+    def _finallize_login(self):
```

Comment:
add return type and fix typo, should be _finalize_login

Reply:
Done